File: src/leetcode/python/BestTimeToBuyAndSellStock_121.py
class BestTimeToBuyAndSellStock_121(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        if not prices or len(prices) == 0:
            return 0
        values = [0] * len(prices)
        for i in range(1, len(prices)):
            values[i] = prices[i] - prices[i-1]
        max_profit, sm = 0, 0
        for i in range(1, len(values)):
            sm += values[i]
            if sm > max_profit:
                max_profit = sm
            elif sm < 0:
                sm = 0
        return max_profit

File: src/leetcode/java/BestTimeToBuyAndSellStock_121.java
package leetcode.java.easy;

public class BestTimeToBuyAndSellStock_121 {
  public int maxProfit(int[] prices) {
    if (prices.length == 0 || prices == null) return 0;

    int maxProfit = 0;
    int[] values = new int[prices.length];
    values[0] = 0;
    for (int i = 1; i < prices.length; i++) {
      values[i] = prices[i] - prices[i - 1];
    }

    int sum = 0;
    for (int i = 1; i < values.length; i++) {
      sum += values[i];
      if (sum > maxProfit) maxProfit = sum;
      else if (sum < 0) sum = 0;
    }
    return maxProfit;
  }
}
